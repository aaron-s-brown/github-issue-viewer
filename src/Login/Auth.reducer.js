import { SET_TOKEN, CLEAR_TOKEN } from '../ActionTypes';

export default function auth(state = {}, action) {
  switch (action.type) {
    case SET_TOKEN:
      return {
        ...state,
        token: action.token
      };
    case CLEAR_TOKEN: 
      return {};
    default:
      return state;
  }
};