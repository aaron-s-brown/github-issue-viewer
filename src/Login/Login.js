import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { SET_TOKEN } from '../ActionTypes';

/*
  Redux actions
*/
const setToken = token => ({
  type: SET_TOKEN,
  token
});

class Login extends React.Component { 

  state = {
    apiKey: '1'
  };

  inputChange = event => {
    this.setState({apiKey: event.target.value});
  };

  handleSubmit= event => {
    this.props.setToken(this.state.apiKey);
    console.log(this.state.apiKey);
    event.preventDefault();
  };
  
  render() {
    return (
    <div className='container'>  
      <form onSubmit={this.handleSubmit}>
        <label htmlFor='apiKey'>Enter your Github API key</label>
        <input type='text' id='apiKey' onChange={this.inputChange} value={this.state.apiKey}></input>
        <input type='submit' value='Submit'></input>
      </form>
    </div>
    );
  }
}

Login.propTypes = {
  setToken: PropTypes.func.isRequired
};

export default connect(
  null,
  {setToken}
)(Login);