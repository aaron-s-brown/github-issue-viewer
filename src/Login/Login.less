.login {
  margin: 1rem 0;

  .login-label {
    display: block;
    text-align: center;
    font-size: 1.5rem;
  }

  input.apiKey {
    display: block;
    margin : .625rem auto;
    width: 60%;
    border-radius: 5px;
    padding: .625rem;
    border: 1px solid #222;
  }

  input.login-submit {
    background-color: #3056ba;
    color: white;
    border: none;
    border-radius: 5px;
    display: block;
    margin: 0 auto;
    padding: 1rem;
    font-size: 1.5rem;
    &:focus {
      outline-offset: 1px;
      outline-style: solid;
      outline-width: 2px;
    }
  }
}