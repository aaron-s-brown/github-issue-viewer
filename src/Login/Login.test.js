import React from 'react';
import {shallow} from 'enzyme';

import {ShallowLogin} from './Login';
import renderer from 'react-test-renderer';


describe('<Login>', () => {

  it('renders correctly', () => {
    const submitMock = jest.fn();
    const tree = renderer
      .create(<ShallowLogin setToken={submitMock}/>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should call setToken prop on form submit', () => {
    const submitMock = jest.fn();
    const login = shallow(<ShallowLogin setToken={submitMock}/>);
    const textInput = login.find('#apiKey').first();
    const form = login.find('form').first();

    textInput.simulate('change', { target: { value: '1a2b3c' } });
    form.simulate('submit', { preventDefault: () => {}});

    expect(submitMock.mock.calls.length).toBe(1);
  })
});