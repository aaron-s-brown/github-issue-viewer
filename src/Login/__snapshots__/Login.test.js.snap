// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login> renders correctly 1`] = `
<div
  className="login"
>
  <form
    onSubmit={[Function]}
  >
    <label
      className="login-label"
      htmlFor="apiKey"
    >
      Enter your Github API key
    </label>
    <input
      className="apiKey"
      id="apiKey"
      onChange={[Function]}
      type="text"
      value=""
    />
    <input
      className="login-submit"
      type="submit"
      value="Submit"
    />
  </form>
</div>
`;
